exercises/joseph_torsney/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,185 @@
+#include <iostream> 
+#include <fstream>
+#include <vector>
+#include <map>
+#include <algorithm>
+#include <string>
+
+/** Checks whether a character is punctuation
+ * 
+ * @param c 
+ * @return true if c is punctuation
+ * @return false otherwise
+ */
+bool myIsPunct(char c) {
+  return(
+    c == '.' ||
+    c == ',' ||
+    c == '?' ||
+    c == '\'' ||
+    c == '\"' ||
+    c == '!' ||
+    c == ')' ||
+    c == '('
+  );
+}
+
+/** Transforms a string to lower case.
+ * 
+ * @param str the string to transform
+ * @return std::string the string in lower case
+ */
+std::string toLowerStr(std::string str) {
+  std::transform(str.begin(), str.end(), str.begin(), 
+    [](unsigned char c){ return std::tolower(c); }
+  );
+
+  return str;
+}
+
+/** Reads the contents of a .txt file at filepath
+ * 
+ * @param filepath the path to the file
+ * @return std::string of the file contents
+ */
+std::string readFile(const std::string& filepath)
+{
+  std::ifstream ifs(filepath);
+  std::string text(
+    (std::istreambuf_iterator<char>(ifs)),
+    (std::istreambuf_iterator<char>())
+  );
+  
+  return(text);
+}
+
+/** Removes punctuation from a string
+ * 
+ * @param str the string to process
+ * @return std::string with no punctuation
+ */
+std::string removePunct(std::string str)
+{
+  for (int i = 0, len = (int) str.size(); i < len; i++) {
```

Comment:
I think this for loop could be made a bit more simple because C++ evaluates the condition before each iteration. So no need to reset len below on L68 and you can just write:
```
for (int i = 0; i < str.size(); i++) {
```
However rather than doing that change you can do something even more concise by using the remove_if function which means you don't need a for loop at all:
```
str.erase(std::remove_if(str.begin(), str.end(), &myIsPunct), str.end());
```